src/dmd/typesem.d

```diff
@@ -3066,7 +3066,10 @@ private extern(C++) final class DotExpVisitor : Visitor
                     return returnExp(null);
 
                 auto die = new DotIdExp(e.loc, alias_e, ident);
-                auto exp = die.semanticY(sc, gagError);
+                /* second argument of semanticY is 2 to signal
+                 * that it was called for an aliasThis
+                 */
+                auto exp = die.semanticY(sc, 2);
```

Comment:
Why can't we just past `1` here to get the same effect?

Reply:
Because the type check in expressionsem may or may not be passed (depending on the type of the alias this) and passing 1 does not affect it anyhow.

Reply:
The fix is to pass 3, because that way we make sure that the first bit is set for other branches.

Reply:
It's ugly, but at least we get rid of the extra parameter.